File: test_regress/t/t_dist_lint_py.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2022 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(dist => 1);

my $root = "..";

if (!-r "$root/.git") {
    skip("Not in a git repository");
} else {
    my $cmd = "cd $root && make lint-py";
    my $out = `$cmd`;
    my $first;
    foreach my $line (split /\n+/, $out) {
        next if $line =~ /^---/;
        next if $line =~ /^flake8/;
        next if $line =~ /^pylint/;
        print "$line\n";
        if (!defined $first) {
            $first = $line;
            error("lint-py failed: ", $first);
        }
    }
}

ok(1);
1;

File: test_regress/t/t_dist_lint_py.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# This program is free software; you can redistribute it and/or modify it
# under the terms of either the GNU Lesser General Public License Version 3
# or the Perl Artistic License Version 2.0.
# SPDX-FileCopyrightText: 2024 Wilson Snyder
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('dist')

if 'VERILATOR_TEST_NO_LINT_PY' in os.environ:
    test.skip("Skipping due to VERILATOR_TEST_NO_LINT_PY")
if not os.path.exists(test.root + "/.git"):
    test.skip("Not in a git repository")

test.run(cmd=["cd " + test.root + " && " + os.environ["MAKE"] + " lint-py"])

test.passes()
